sds/src/scanner/suppression.rs

```diff
@@ -0,0 +1,68 @@
+use ahash::AHashSet;
+use serde::{Deserialize, Serialize};
+use serde_with::serde_as;
+
+#[serde_as]
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+pub struct SuppressionConfig {
+    #[serde(default)]
+    pub starts_with: Vec<String>,
+    #[serde(default)]
+    pub ends_with: Vec<String>,
+    #[serde(default)]
+    pub exact_match: Vec<String>,
+}
+
+pub struct CompiledSuppressionConfig {
+    pub starts_with: Vec<String>,
+    pub ends_with: Vec<String>,
+    pub exact_match: AHashSet<String>,
+}
+
+impl CompiledSuppressionConfig {
+    pub fn should_match_be_suppressed(&self, match_content: &str) -> bool {
+        let exact_match = self.exact_match.contains(match_content);
```

Comment:
I know I suggested this, but I realized this removed the short-circuit so you are still checking all 3 even if one is found.

It's probably better to individually short-circuit to keep it easy to read

```rust
if self.exact_match.contains(match_content) {return true;}

if self
    .starts_with
    .iter()
    .any(|start| match_content.starts_with(start)) {return true;}

if self
    .ends_with
    .iter()
    .any(|end| match_content.ends_with(end)) {return true'}

false
```